src/asyncresults.jl

```diff
@@ -0,0 +1,242 @@
+"An asynchronous PostgreSQL query"
+mutable struct AsyncResult
+    "The LibPQ.jl Connection used for the query"
```

Comment:
out of interest, why document line-by-line rather than putting this in the docstring 
```julia
"""
    AsyncResult

An asynchronous PostgreSQL query.

# Fields
- `jl_conn::Connection`: The LibPQ.jl Connection used for the query.
- etc...
"""

Reply:
Using DocStringExtensions, that's what happens! :)

```
# Docstring template for types using DocStringExtensions
@template TYPES =
    """
        $(TYPEDEF)

    $(DOCSTRING)

    ## Fields:

    $(FIELDS)
    """
```

Reply:
phwoarr

we _can_ have nice things